extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/devmode/ReplacementDebugPage.java

```diff
@@ -16,113 +16,142 @@
 
 package io.quarkus.undertow.deployment.devmode;
 
-import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
 
 import io.undertow.server.HttpServerExchange;
 import io.undertow.util.Headers;
 
 /**
- * generates a error page with a stack trace
+ * Generates an error page with a stack trace
  *
  * @author Stuart Douglas
  */
 public class ReplacementDebugPage {
 
-    public static final String ERROR_CSS = "<style>\n" +
+    private static final String HTML_TEMPLATE = "" +
+            "<!doctype html>\n" +
+            "<html lang=\"en\">\n" +
+            "<head>\n" +
+            "    <title>Error restarting Quarkus: %1$s</title>\n" +
+            "    <meta charset=\"utf-8\">\n" +
+            "    <style>%3$s</style>\n" +
+            "</head>\n" +
+            "<body>\n" +
+            "" +
+            "<header>\n" +
+            "    <h1 class=\"container\">Error restarting Quarkus</h1>\n" +
+            "    <div class=\"exception-message\">\n" +
+            "        <h2 class=\"container\">%1$s</h2>\n" +
```

Comment:
Surely this should have some kind of `pre` + `code` style formatting?  Otherwise, as in the example, the `^]` seems very out of context.

Reply:
@maff I think I would display the first line only in the title. That fixes David's remark too.

Otherwise I think it looks very good.

Reply:
`pre`/`code` would lead to a wide/scrolling header area which I wanted to avoid as the exception message is properly formatted in the stack trace box below. Trimming the message to the first line sounds good. I updated the PR accordingly.